File: 0078-subsets/0078-subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        result=[]
        current=[]
        self.backtracking(0,nums,result,current)
        return result
    def backtracking(self,i:int,nums:List[int],result:List[List[int]],current:List[int]):
        if i>=len(nums):
            result.append(current[:])
            return
        self.backtracking(i+1,nums,result,current)
        current.append(nums[i])
        self.backtracking(i+1,nums,result,current)
        current.pop()
       
        

File: 0078-subsets/0078-subsets.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>>result;
        vector<int>current;
        backtracking(0,nums,current,result);
        return result;
        
    }
    void backtracking(int i,vector<int>&nums,vector<int>&current,vector<vector<int>>& result ){
        
        if(i>=nums.size()){
            result.push_back(current);
            return;

        }
        backtracking(i+1,nums,current ,result);
        current.push_back(nums[i]);
         backtracking(i+1,nums,current ,result);
        current.pop_back();
       
    }
};